platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ConfigurationRoleUsageIntegrationTest.groovy

```diff
@@ -420,57 +429,143 @@ class ConfigurationRoleUsageIntegrationTest extends AbstractIntegrationSpec impl
+        when:
+        executer.expectDocumentedDeprecationWarning("The configuration customCompileOnly was created explicitly. This configuration name is reserved for creation by Gradle. This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 9.0. Do not create a configuration with the name customCompileOnly. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#configurations_allowed_usage")
+        executer.expectDocumentedDeprecationWarning("Gradle will mutate the usage of configuration customCompileOnly to match the expected usage. This may cause unexpected behavior. Creating configurations with reserved names has been deprecated. This will fail with an error in Gradle 9.0. Create source set custom prior to creating or accessing the configurations associated with it. For more information, please refer to https://docs.gradle.org/current/userguide/building_java_projects.html#sec:implicit_sourceset_configurations in the Gradle documentation.")
         fails 'help'
 
+        then:
+        failure.assertHasDescription("An exception occurred applying plugin request [id: 'java']")
+        failure.assertHasCause("Failed to apply plugin class 'org.gradle.api.plugins.JavaBasePlugin'.")
+        failure.assertHasCause("Gradle cannot mutate the usage of configuration 'customCompileOnly' because it is locked.")
+    }
+
+    def "changing usage #property = #change on detached configurations fails"() {
+        given:
+        buildFile << """
+            def detached = project.configurations.detachedConfiguration()
+
+            assert !detached.canBeConsumed
+            assert detached.canBeResolved
+            assert detached.canBeDeclared
+
+            detached.$property($change)
+        """
+
+        when:
+        fails "help"
+
         then:
         failure.assertHasDescription("A problem occurred evaluating root project '${buildFile.parentFile.name}'.")
         failure.assertHasCause("""Method call not allowed
-  Calling setCanBeResolved(true) on configuration ':$configuration' is not allowed.  This configuration's role was set upon creation and its usage should not be changed.""")
+  Calling $property($change) on configuration ':detachedConfiguration1' is not allowed.  This configuration's role was set upon creation and its usage should not be changed.""")
 
-        where: "a non-exhaustive list of configurations is tested"
-        configuration << ['api', 'implementation', 'compileOnly', 'runtimeOnly', 'archives']
+        where:
+        property | change
+        "setCanBeConsumed" | true
+        "setCanBeResolved" | false
+        "setCanBeDeclared" | false
     }
 
-    def "setting consumable = false fails for consumable configurations added by java plugin"() {
-        given: "a buildscript which attempts to change a configuration's usage"
+    def "changing usage on detached configurations fails when flag is set"() {
```

Comment:
Moving these tests around in the file has made it quite difficult to compare the diff between the old and new behavior. I am having trouble determining if test cases have been removed or just moved elsewhere

I understand we have warning/failure regions, but I would say an easier to verify diff is more important than keeping these regions intact.

Reply:
Alright, I'll revert the test order to match, and then add a final commit to this PR that only moves them to the appropriate regions, so we can verify the contents and the moves separately.